internal/transport/grpchttp2/errors.go

```diff
@@ -70,3 +70,9 @@ func (err ErrCode) String() string {
 	}
 	return fmt.Sprintf("unknown error code %#x", uint32(err))
 }
+
+type connError ErrCode
+
+func (err connError) Error() string {
+	return fmt.Sprintf("connection error: %s", ErrCode(err))
+}
```

Comment:
Do we really need this new type? Can't we define the `Error` method on `ErrCode`?

Reply:
The main reason was having a difference between a connection error and a stream error. The stream error would be introduced when the new framer is created. They both use the same error codes. What do you think about this approach?

Reply:
At this point, you are using `connError` type to return an error from `ReadFrame` *after* you have successfully read from the frame, but you don't support the frame type. Why is this considered to be a connection error? How will stream errors be different? Can you clearly explain the difference between the two types of errors and when each of them would be used? Thanks.